cultivation/reader_app/main.py

```
@@ -0,0 +1,267 @@
+            raise ValueError(f"Metadata schema validation failed: {ve}")
+    return data
+
+# API: GET /metadata/{arxiv_id} (returns validated metadata)
+from fastapi import HTTPException
+@app.get("/metadata/{arxiv_id}")
+def get_metadata(arxiv_id: str):
+    try:
+        data = load_and_validate_metadata(arxiv_id)
+        return JSONResponse(content=data)
+    except FileNotFoundError as e:
+        raise HTTPException(status_code=404, detail=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)